grails-app/domain/org/pih/warehouse/order/OrderAdjustment.groovy

```diff
@@ -111,4 +111,9 @@ class OrderAdjustment implements Serializable {
     Boolean getHasRegularInvoice() {
         return invoices.any { it.invoiceType == null || it.invoiceType?.code == InvoiceTypeCode.INVOICE }
     }
+
+    @Override
+    int compareTo(OrderAdjustment o) {
+        return dateCreated <=> o.dateCreated
```

Comment:
Careful with this. This would disappear adjustments with the same dateCreated that are added to a TreeSet or TreeMap. 

> Note that the ordering maintained by a set (whether or not an explicit comparator is provided) must be consistent with equals if it is to correctly implement the Set interface. (See Comparable or Comparator for a precise definition of consistent with equals.) This is so because the Set interface is defined in terms of the equals operation, but a TreeSet instance performs all element comparisons using its compareTo (or compare) method, so two elements that are deemed equal by this method are, from the standpoint of the set, equal. The behavior of a set is well-defined even if its ordering is inconsistent with equals; it just fails to obey the general contract of the Set interface.

https://docs.oracle.com/javase/8/docs/api/java/util/TreeSet.html

Reply:
I usually do something like this 
```
    @Override
    int compareTo(OrderAdjustment obj) {
        return dateCreated <=> obj?.dateCreated ?:
                lastUpdated <=> obj?.lastUpdated ?:
                        id <=> party.id
    }
```